File: Easy/Horse-racing-duals/Horse-racing-duals.java
import java.util.*;

class Solution {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int N = in.nextInt();
        List<Integer> ps = new ArrayList<>(N);
        for (int i = 0; i < N; i++) {
            ps.add(in.nextInt());
        }
        ps.sort(null);
        int small = Integer.MAX_VALUE;
        for (int i = 0; i < N - 1; i++) {
            small = Math.min(small, Math.abs(ps.get(i) - ps.get(i + 1)));
        }
        System.out.println(small);
    }
}

File: Easy/Horse-racing-duals/Horse-racing-duals.swift
import Glibc
import Foundation

public struct StderrOutputStream: TextOutputStream {
    public mutating func write(_ string: String) { fputs(string, stderr) }
}
public var errStream = StderrOutputStream()

let N = Int(readLine()!)!
var ps : [Int] = Array()
for i in 0..<N {
    ps.append(Int(readLine()!)!)
}
ps.sort()
var small = Int.max
for i in 0..<(ps.count-1) {
    small = min(small, abs(ps[i] - ps[i + 1]))
}
print(small)
